app/src/main/java/com/portfolio/tasky/entry/EntryActivity.kt

```diff
@@ -2,28 +2,75 @@ package com.portfolio.tasky.entry
 
 
 import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import android.view.View
+import android.view.animation.TranslateAnimation
 import androidx.appcompat.app.AppCompatActivity
-import androidx.appcompat.widget.Toolbar
+import androidx.appcompat.widget.AppCompatTextView
+import androidx.core.content.ContextCompat
 import androidx.databinding.DataBindingUtil
 import com.portfolio.tasky.*
 import com.portfolio.tasky.databinding.ActivityEntryBinding
 import com.portfolio.tasky.entry.fragments.LoginFragment
-import com.portfolio.tasky.usecases.FragmentInflater
-import com.portfolio.tasky.usecases.FragmentInflaterImpl
-import com.portfolio.tasky.usecases.ToolbarHandler
-import com.portfolio.tasky.usecases.ToolbarHandlerImpl
+import com.portfolio.tasky.globals.Constants
+import com.portfolio.tasky.usecases.*
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
 class EntryActivity : AppCompatActivity(), FragmentInflater by FragmentInflaterImpl(), ToolbarHandler by ToolbarHandlerImpl() {
-    private var viewBinding: ActivityEntryBinding? = null
+    private lateinit var viewBinding: ActivityEntryBinding
+    private lateinit var networkChangeReceiver : NetworkChangeReceiver
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         viewBinding = DataBindingUtil.setContentView(this, R.layout.activity_entry)
 
+        setObservers()
         startLoginFragment()
     }
 
+    private fun setObservers() {
+        networkChangeReceiver = NetworkChangeReceiver(this)
+        networkChangeReceiver.observe(this){
+            when(it){
+                NetworkStatus.Available -> networkAvailability(true)
+                NetworkStatus.Unavailable -> networkAvailability(false)
+            }
+        }
+    }
+
+    private fun networkAvailability(isAvailable : Boolean){
```

Comment:
Similar comment as in a previous PR: I'd keep function names as clear as possible, so here I'd rather use `onNetworkAvailabilityChanged()` or similar

Reply:
Sorry for this. Currently I am desperately investing my thought process in understanding the architecture and that's why compromising it in these places. I'll try to remember to provide proper names in future PRs.

Done